docs/source/sci_api_tutorial/armer.rst

```diff
@@ -2,21 +2,297 @@
+        path for submission script generation.
+        (default: sub_dir/submit.cmd; will be sub_dir/submit_#.cmd if the file exists. # is a growing index)
+           
+
+3. Dynamic Monitoring
+------------------------------------------------
+    Once the job has been submitted, a job ID is added to the object by the function. By tracing the job ID, the “workflow script” can monitor the status of a job object in the queue, and mediate the status by killing, holding, or releasing the job.
+
+.. dropdown:: :fa:`eye,mr-1` Click to learn more about `Dynamic monitoring`
+    
+    The capability of dynamically monitoring the job completion status is vital to high-throughput modeling workflow because the workflow involves multiple different types of simulation subtasks that must be sequentially operated.
+    
+    Two methods have been implemented to achieve dynamic monitoring, they are: wait_to_end() and wait_to_array_end() methods. The wait_to_end() method checks the status of a job in the job queue within a certain period of time (i.e., every 30 s) and exits upon the detection of messages that indicate job completion, error, or cancellation. The wait_to_array_end() method takes multiple job objects and submits them in one job array. Similarly, this method also monitors the status of all jobs in the array regularly and dynamically appends new jobs to the array up to the maximal capacity (i.e., array size).
+
+.. dropdown:: :fa:`eye,mr-1` Click to learn more about the arguments in Dynamic monitoring
+    
+    There are two functions: "wait_to_end" (single submission) and "wait_to_array_end" (array submission)
+        
+    ``period``
+        the time cycle for update job state change (Unit: s)
+    
+    The following arguments are array submission only:
+
+    ``jobs``
+        a list of ClusterJob object to be execute
+        
+    ``array_size``
+        
+        how many jobs are allowed to submit simultaneously. 
+
+        (e.g. 5 for 100 jobs means run 20 groups. All groups will be submitted and 
+        in each group, submit the next job only after the previous one finishes.)
+        
+    ``sub_dir``
+        (default: self.sub_dir)
+
+        submission directory for all jobs in the array. 
+        
+        Overwrite existing self.sub_dir in the job obj
+        
+        * you can set the self value during config_job to make each job different
+    
+    ``sub_scirpt_path`` 
+        (default: self.sub_script_path)
+        
+        path of the submission script. Overwrite existing self.sub_script_path in the job obj
+        
+        * you can set the self value during config_job to make each job different
+
+
+
+For most users, only the `Job configuration` requires specific input from different users.
+The first step involves configuring a job using the config_job method. This method prepares the submission script by specifying the commands to be executed, environmental settings, and resource requirements.
```

Comment:
Change this Bold and change it to say

none of the function will be directly access by user in normal EnzyHTP workflows. They are used by developers in APIs of EnzyHTP. However, all of the APIs that uses ARMer will expose an input that allow users to config the resource ...(elaborate)...

Reply:
And in the next section, you can smoothly introduce the cluster_job_config dict